lib/beaker/dsl/helpers/web_helpers.rb

```diff
@@ -52,9 +53,17 @@ def fetch_http_file(base_url, file_name, dst_dir)
           else
             logger.notify "Fetching: #{src}"
             logger.notify "  and saving to #{dst}"
-            open(src, :allow_redirections => :all) do |remote|
-              File.open(dst, "w") do |file|
-                FileUtils.copy_stream(remote, file)
+            begin
+              open(src, :allow_redirections => :all) do |remote|
+                File.open(dst, "w") do |file|
+                  FileUtils.copy_stream(remote, file)
+                end
+              end
+            rescue OpenURI::HTTPError => e
+              if e.message =~ /404.*/
```

Comment:
Is the status code available at `e.code`?  I can't quite tell from ruby docs, it's not exactly clear; didn't dig into the source code either.